lib/core/src/animation/differential_time_animator.rs

```diff
@@ -0,0 +1,47 @@
+use super::ContinuousTimeAnimator;
+use super::FnAnimation;
+
+
+
+// ====================================
+// === DifferentialTimeAnimatorData ===
+// ====================================
+
+struct DifferentialTimeAnimatorData {
+    closure          : Box<dyn FnMut(f32)>,
+    previous_time    : Option<f32>
+}
+
+impl DifferentialTimeAnimatorData {
+    pub fn new<F:FnAnimation>(f:F) -> Self {
+        let closure          = Box::new(f);
+        let previous_time    = None;
+        Self { closure,previous_time }
+    }
+}
+
+
+
+// ================================
+// === DifferentialTimeAnimator ===
+// ================================
+
+/// This structure runs an animation every frame with the time difference from the last frame as
+/// its input.
+pub struct DifferentialTimeAnimator {
```

Comment:
Are you sure you wanted to use the "differential" word here? You are not computing any differentials there, as far as I see.
https://en.wikipedia.org/wiki/Differential_(mathematics)
https://en.wikipedia.org/wiki/Differential_of_a_function

Reply:
We agreed renaming `DifferentialTimeAnimator` to `Animator`.